sdk/keyvault/azure-security-keyvault-secrets/src/test/java/com/azure/security/keyvault/secrets/KeyVaultCredentialPolicyTest.java

```diff
@@ -126,10 +137,53 @@ public void onChallengeCredentialPolicy() {
     public void onAuthorizeRequestChallengeCachePresent() {
         KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(this.credential, false);
 
+        StepVerifier.create(onChallenge(policy, this.callContext, unauthorizedHttpResponseWithHeader) // Challenge cache created
+                .then(policy.authorizeRequest(this.testContext))) // Challenge cache used
+            .verifyComplete();
+
+        String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(HttpHeaderName.AUTHORIZATION);
+        assertFalse(tokenValue.isEmpty());
+        assertTrue(tokenValue.startsWith(BEARER));
+    }
+
+    @Test
+    public void onAuthorizeRequestChallengeCachePresentSync() {
+        KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(this.credential, false);
+
         // Challenge cache created
-        onChallenge(policy, this.callContext, unauthorizedHttpResponseWithHeader).block();
+        onChallengeSync(policy, this.callContext, unauthorizedHttpResponseWithHeader);
         // Challenge cache used
-        policy.authorizeRequest(this.testContext).block();
+        policy.authorizeRequestSync(this.testContext);
+
+        String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(HttpHeaderName.AUTHORIZATION);
+        assertFalse(tokenValue.isEmpty());
+        assertTrue(tokenValue.startsWith(BEARER));
+    }
+
+    @Test
+    public void onAuthorizeRequestChallengeCachePresentWithClaims() {
+        KeyVaultCredentialPolicy policy = new KeyVaultCredentialPolicy(this.credential, false);
+
+        StepVerifier.create(onChallenge(policy, this.callContext, unauthorizedHttpResponseWithHeader) // Challenge cache created
+                .then(onChallenge(policy, this.callContext, unauthorizedHttpResponseWithHeader)) // Challenge with claims received
+                .then(policy.authorizeRequest(this.testContext))) // Challenge cache used
+            .verifyComplete();
+
+        String tokenValue = this.testContext.getHttpRequest().getHeaders().getValue(HttpHeaderName.AUTHORIZATION);
+        assertFalse(tokenValue.isEmpty());
```

Comment:
We should also check that the policy requests a CAE token, passes the correct claims to the credential, and doesn't pass claims when the response doesn't include them